webpack/scenes/ContentViews/Details/ContentViewDetails.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { useSelector, shallowEqual } from 'react-redux';
+import { Grid, GridItem, TextContent, Text, TextVariants } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import PropTypes from 'prop-types';
+
+import DetailsContainer from './DetailsContainer';
+import ContentViewInfo from './ContentViewInfo';
+import { selectCVDetails } from './ContentViewDetailSelectors';
+import TabbedView from '../../../components/TabbedView/TabbedView';
+
+const ContentViewDetails = ({ match }) => {
+  const cvId = parseInt(match.params.id, 10);
+  const details = useSelector(state => selectCVDetails(state, cvId), shallowEqual);
+  const {
+    name,
+    label,
+    description,
+    composite,
+    force_puppet_environment: forcePuppetEnvironment,
+    solve_dependencies: solveDependencies,
+  } = details;
+  const info = {
+    name: {
+      value: name,
+      label: __('Name'),
+    },
```

Comment:
Took me 15+ minutes to get into my head how this object literal ends up as JSX.  It was confusing to me (for some reason) that the object keys end up as `key`s, though that actually makes sense.  Even so, what you're doing here looks _very_similar to JSX markup assigning different props to each child, but it's in the object literal form so it's hard to comprehend.

Is it possible to do some refactoring of `ContentViewInfo.js` and possibly turn all of these inner objects into props that you assign to React components?  I just think that down the line it would make it that much easier to comprehend and maintain.

Reply:
Refactoring in this way would also allow you to eliminate some of the confusion I had around "why does this object have different keys than the previous one?" by using PropTypes and default props.

Reply:
agreed, this started out simple but got a little too complicated. I updated this to use a new `ActionableDetail` component instead of iterating over this object